### PR TITLE
Cash Byte App completed

### PR DESCRIPTION
**Pull request by Metaverse-Nabeel**
### The following changes have been made to this PR:

**Day 1**

- [x] Set up the repository and tools.
- [x] Create your models according to the ERD diagram. Don't forget about validations and tests.
- [x] Implement the authentication and authorization.

**Day 2**

- [x] Create the page to add a category.
- [x] Create the home page.
- [x] Create the page to add a transaction.
- [x] Create the transactions page.

**Day 3**

- [x] Make sure your app is tested adequately.
- [x] Deploy the project, and test for final details.
- [x] Record a video for your project.
- [x] Create a good README and PR description.

**Technical requirements**

- [x] I used Postgres as your database.
- [x] I used devise for authentication.
- [x] I validated all user input to make sure that anyone with bad intentions cannot compromise your app.
- [x] I used a view template engine of .erb
- [x] The project can not be deployed and is not accessible online due to some encryption errors I have tried a lot. If you can help me with the deployment then ask for revision otherwise please complete it. I have already taken help from many other developer.

### CODE REVIEW

**Review by Whoistolu — COMMENTED** [review]
Hi @Metaverse-Nabeel 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project, that its linters are not running. Kindly resolve this. Also, you attached the wrong links for the deployment and the presentation video. Kindly ensure to resolve this as well. Lastly, your MIT link is broken, kindly fix this so your project will be following the proper requirements.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Metaverse-Nabeel** [comment]
Hi @Whoistolu,

Thank you so much for your feedback. I have made the required changes. Now, I will again ask for a review.

Regards

**Review by nahnahdev — COMMENTED** [review]
## Project Status: INVALIDATED :red_circle: 
Hi @Metaverse-Nabeel , 

While you made a great effort in this project, I can't go ahead and review your code.


## Invalid Code Review Request

- This is because you have submitted a project, that has an error that is preventing me from runnin your app successfully. So kindly fix this :+1: It seems to be coming from your `manifest.js` file :+1: 
- Error message
image


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Metaverse-Nabeel** [comment]
Hello @NahnahAJ,

I hope you are doing well. I saw that you have made this PR invalidated but I found everything working perfectly fine. 
 Here is the proof that everything works fine. [Video](https://www.loom.com/share/36d572d20d244bfb941b2fcd33092ce2?sid=9b9e62e8-1500-4f1b-b5c0-f46f1cb49fbd) 
 
Please review it again. 

Thank you

File: README.md [hunk 1]
```diff
@@ -76,7 +76,7 @@
 
 ## 🚀 Presenation video <a name="live-demo"></a>
 
-- [Video](https://www.loom.com/share/c618a0f2f5ff4b5fb5759fe4181fea21?sid=2ff4de93-69d9-44c0-b631-275a44d9d258)
+- [Video](https://www.loom.com/share/2f0d707ba4f24105999dff5f9ce26b84?sid=2006c4a2-f8e2-48cf-bba6-f4767db1a198)
```
Comment (cynthiainga) [inline README.md]: - [x] Good job so far but please kindly highlight about a piece of the code that you're proud of as mentioned in the requirement and your video will be completed 🙃 😁 

Screenshot 2023-08-11 at 22 02 47

- [x] Also the live demo link should be present in the README file, kindly do your best to deploy your project and add the live demo link 🙏🏾 
- [x] The MIT License link is broken kindly fix that, having broken links in the README file doesn’t look more professional 😎 😎 

Screenshot 2023-08-11 at 22 07 13

File: spec/requests/categories_spec.rb
Comment (cynthiainga) [inline spec/requests/categories_spec.rb:1]: - [x] Please add tests for this file and the categories in the model folder, you just create the user and category but you didn’t write the tests. Kindly fix that 🙏🏾

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @Metaverse-Nabeel, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [x] No linters errors ✔️ 
- [x] Use Postgres as the database ✔️ 
- [x] Descriptive PR ✔️ 

## Required Changes ♻️

- [x] I left my comment here in my summary because almost all the concerned files are not present in this PR. Kindly make sure to use the correct colors given in the design 🙃 
Screenshot 2023-08-11 at 22 48 56

- [x] Please kindly make sure your registration page match to the original design 👇🏾 

> |     | your design                       | given design                    |     |
> | --- | ----------------------------------- | ------------------------------------ | --- |
> |     | Screenshot 2023-08-11 at 22 54 21 | Screenshot 2023-08-11 at 22 18 18

- [x] It would be better to follow the design and make sure the category page match to the original design 'colors,...'

> |     | your design                       | given design                    |     |
> | --- | ----------------------------------- | ------------------------------------ | --- |
> |     | Screenshot 2023-08-11 at 23 05 23 | Screenshot 2023-08-11 at 23 05 47

- [x] Also kindly make sure that the details page match to the original design (eg: colors, padding, margin,...)

> |     | your design                       | given design                    |     |
> | --- | ----------------------------------- | ------------------------------------ | --- |
> |     | Screenshot 2023-08-11 at 23 30 27 | Screenshot 2023-08-11 at 23 31 42

- [x] Please kindly add `confirmable` to your `devise` for the email confirmation, the application should ask for confirmation of email 😁 😁 

- [x] Kindly check the comments under the review for other changes 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @Metaverse-Nabeel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Metaverse-Nabeel** [comment]
Hello @fobadara 

I hope you are doing well. Thank you so much. I am obliged.

Best Regards

**Pull request merged**